scan-payment/src/main/java/com/getbouncer/scan/payment/ml/AlphabetDetect.kt

```diff
@@ -0,0 +1,111 @@
+import com.getbouncer.scan.framework.image.toRGBByteBuffer
+import com.getbouncer.scan.framework.ml.TFLAnalyzerFactory
+import com.getbouncer.scan.framework.ml.TensorFlowLiteAnalyzer
+import com.getbouncer.scan.framework.util.scaleAndCenterWithin
+import com.getbouncer.scan.payment.R
+import java.io.FileNotFoundException
+import java.nio.ByteBuffer
+import org.tensorflow.lite.Interpreter
+
+private val TRAINED_IMAGE_SIZE = Size(48, 48)
+
+/** model returns whether or not there is a screen present */
+private const val NUM_CLASS = 27
+class AlphabetDetect private constructor(interpreter: Interpreter) :
+    TensorFlowLiteAnalyzer<AlphabetDetect.Input, ByteBuffer,
+            AlphabetDetect.Prediction,
+            Array<FloatArray>>(interpreter) {
+
+    companion object {
+        /**
+         * Calculate what portion of the full image should be cropped for screen detection.
+         */
+        fun calculateCrop(fullImage: Size): Rect =
+            Size(9, 16).scaleAndCenterWithin(fullImage) // Expects 16:9 aspect ratio
+    }
+
+    data class Input(val objDetectionImage: Bitmap)
+
+    data class Prediction(val character: Char, val confidence: Float, val probabilities: FloatArray)
+
+    override val name: String = Factory.NAME
+
+    override fun buildEmptyMLOutput() = arrayOf(FloatArray(NUM_CLASS))
+
+    override fun interpretMLOutput(data: Input, mlOutput: Array<FloatArray>): Prediction {
+        val prediction = mlOutput[0]
+        val index = getIndexOfLargest(prediction)
+        var c = ' '
+        if (index != 0) {
+            c = ('A'.toInt() - 1 + index).toChar()
+        }
+        return Prediction(
+            c , prediction[index], prediction
+        )
+    }
+
+    private fun getIndexOfLargest(array: FloatArray?): Int {
+        if (array == null || array.size == 0) return -1 // null or empty
+        var largest = 0
+        for (i in 1 until array.size) {
+            if (array[i] > array[largest]) largest = i
+        }
+        return largest // position of the first largest found
+    }
+
+
+    override fun transformData(data: Input): ByteBuffer = data.objDetectionImage
+        //.crop(calculateCrop(data.objDetectionImage.size()))
+        .scale(TRAINED_IMAGE_SIZE)
+        .toRGBByteBuffer()
+
+    override fun executeInference(
+        tfInterpreter: Interpreter,
+        data: ByteBuffer,
+        mlOutput: Array<FloatArray>
+    ) = tfInterpreter.run(data, mlOutput)
+
+    /**
+     * A factory for creating instances of the [ScreenDetect]. This downloads the model from the
+     * web. If unable to download from the web, this will throw a [FileNotFoundException].
+     */
+    class Factory(context: Context, loader: Loader) : TFLAnalyzerFactory<AlphabetDetect>(loader) {
+        companion object {
+            private const val USE_GPU = false
+            private const val NUM_THREADS = 2
+            const val IS_THREAD_SAFE = true
+
+            const val NAME = "name_detect"
```

Comment:
Please make sure to add this to the ScanStats.

Reply:
is there a generic way to do this?

Reply:
we don't have encrypted payloads for cardscan, are there other stats we should add?

Reply:
Should be fine